### DISCUSSION TITLE
Redis: Warning `Memory overcommit must be enabled`

### DISCUSSION THREAD

**Discussion by ManOki** [nextcloud/all-in-one#1731] [Wiki]
```
WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition.
```

If you see a warning like this, you have to enable `overcommit` on your **host** machine (outside docker).

### Solution

- `echo "vm.overcommit_memory = 1" | sudo tee /etc/sysctl.d/nextcloud-aio-memory-overcommit.conf` should be set, to enable Memory overcommit on reboot, change will apply after reboot or see the line below how to activate it temporary without reboot
- `sysctl "vm.overcommit_memory=1"` enables Memory overcommit on-the-fly temporary, then restart Nextcloud AIO from the AIO Interface

**Note for Docker Desktop instances**: See https://stackoverflow.com/a/69294687 for how to apply sysctl config values.

### Background
Due to the fact that (for good reasons) the Redis container does not run in privileged mode, there is no way that Nextcloud AIO or Redis can enable this for you, i.e., `/proc` is read-only for containers.

See [Redis documentation](https://redis.io/docs/getting-started/faq/#background-saving-fails-with-a-fork-error-on-linux), corresponding [docker issue](https://github.com/docker-library/redis/issues/19#issuecomment-96080885) and [kernel documentation](https://www.kernel.org/doc/Documentation/vm/overcommit-accounting)

**Comment by szaimen** [comment]
Wow, thanks a lot for this nice guide! :)

**Reply by alkhachatryan** [reply]
it helped me with laravel, psql and redis stack

**Comment by loren-klemesrud-gridbeyond** [comment]
This was so helpful!

**Comment by NiciTheNici** [comment]
Thank you! This has been an issue for a while for me <3

**Comment by Irdiism** [comment]
Just noticed the issue on one of my Nextcloud instances and solved it in a couple of minutes. Thanks a lot for your contribution!

**Comment by woiwoiwoiwoi** [comment]
Unfortunately, I can't get this to work, and am thankful for any pointers.

I hope I'm correct in understanding that I am to run these commands on my host system? My host system is a QNAP NAS. The first command results in `No such file or directory`, so I created `/etc/sysctl.d` and re-ran the command, which succeeded. Then I ran the second command, to which the response is `sysctl: error: 'vm.overcommit_memory=1' is an unknown key`. I then tried the instructions from the redis log, which are slightly different, and added `'vm.overcommit_memory = 1'` to `/etc/sysctl.conf`, which produced the same error message.

What can I do to fix this?

Also, is this a rare misconfiguration on some systems (like mine) or is this a common error message that most people don't notice since it's buried in the redis logs?

**Reply by EVOTk** [reply]
Hello,

I think you need to create the file manually on a QNAP NAS. This command should allow you to do that: 
`echo 1 > /proc/sys/vm/overcommit_memory`

QTS will delete this file each time the NAS is booted or restarted, but you can solve this problem by including it in startup.sh : https://www.qnap.com/en/how-to/faq/article/running-your-own-application-at-startup

**Reply by woiwoiwoiwoi** [reply]
Fantastic, this works, thank you!

Regarding autorun.sh, I'll report back once I've tested it properly.
